Update PULL_REQUEST_TEMPLATE.md based on new breaking change definitions (#11)
terraform-ibm-modules/.github: .github/PULL_REQUEST_TEMPLATE.md

### Types of changes in this PR

> #### Change(s) that impact the core Terraform module and/or any submodules

Reviewer: ```suggestion
#### Changes that affect the core Terraform module or submodules
```

- [ ] Bug fix
- [ ] New feature
- [ ] Dependency update

> #### Change(s) that don't affect the core Terraform module and/or any submodules

Reviewer: ```suggestion
#### Changes that don't affect the core Terraform module or submodules
```

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?

> See the following [doc](https://terraform-ibm-modules.github.io/documentation/#/versioning) which defines what types of changes should generate what kind of SemVer release, and tick the appropriate box below:

Reviewer: ```suggestion
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).
```

- [ ] No release
- [ ] Patch release (`x.x.X`))

> - [ ] Minor release (`x.X.x`))

Reviewer: ```suggestion
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
```

- [ ] Major release (`X.x.x`)

##### Release notes content

> If a release is required, replace this text with information that users need to know about the release. Word it in such a way that makes sense the consumer, and ensure to include any changes necessary to update to the new version from the previous one. This information helps the merger to prepare the commit message that will be used when squash and merging the PR, which eventually ends up being published in the release notes for the module.

Reviewer: ```suggestion
If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version. 

Your notes helps the merger write the commit message for the PR that is published in the release notes for the module.
```

Trying to simplify and shorten the text so people might actually scan it.
